### DISCUSSION TITLE
Generated Prisma Repository's FindUnique() and FindFirst() always returns a "valid" object

### DISCUSSION THREAD

**Discussion by LefanTan** [TypedProject/ts-express-decorators#2200] [Q&A] [answered]
Hi folks.

Whenever I use the generated repository and use its `findFirst` or `findUnique` function to find a record that doesn't exist, it returns an object as follow, instead of null as the return type of the function suggested. 

```
{
    "name": "NOT_FOUND",
    "status": 404,
    "message": "User not found"
}
```

This causes some unexpected bug when implementations are expecting null object to throw the appropriate error, see example below

UserService.ts
```
import { Injectable } from "@tsed/di";
import { UserModel, UsersRepository } from "@schema";

@Injectable()
export class UserService extends UsersRepository {

  /**
   * Find user by Id
   * @param id
   */
  async findUserById(id: string) {
    return await this.findUnique({
      where: {
        id,
      },
    });
  }
}

```

UserController.ts

```
  @Get("/:uid")
  @Description("Return a specified user")
  @Returns(200, UserModel)
  async get(@PathParams("uid") uid: string) {
    try {
      const user = await this.service.findUserById(uid);

      // user is always a "valid" object, and hence this statement evaluate to true.
      if (!user) return new NotFound("User not found");
      return user;
    } catch (e) {
      throw new BadRequest(e);
    }
  }
```
Alternatively in UserService.ts, I can use this.collection.findFirstOrThrow, and deserialize it to the tsed models. But I'm curious if this could be a bug

**Comment by LefanTan** [comment, marked as answer]
This is a silly mistake and not related to tsED. That happens because I threw an error in the try clause and catches it again in the catch clause. 

I would remove this discussion if i can.

**Reply by Romakita** [reply]
Hello @LefanTan 
Also the generated repository is a proxy of the prisma repository ^^. The result is directly return by prisma itself ;)

see you !

**Reply by Romakita** [reply]
Ref: https://github.com/tsedio/tsed/blob/b3d3be6912ed75f3e43f296b6e041e73e984a4bc/packages/orm/prisma/src/generator/utils/generateRepositories.ts#L136
